File: DataStructuresAndAlgorithms/1. Array and Hashing/3. Two Sum/twoSum.py
from collections import Counter

def twoSum(nums: list, target: int) -> list:
    counts = Counter(nums)
    for i in range(len(nums)):
        counts[nums[i]] -= 1
        anVal = target - nums[i]
        if anVal in counts and counts[anVal] > 0:
            return [i, nums.index(anVal, i+1)]

if __name__ == "__main__":
    # nums = [3,2,4]
    # target = 6
    nums = [2,7,11,15]
    target = 9
    print(twoSum(nums, target))

File: DataStructuresAndAlgorithms/1. Array and Hashing/3. Two Sum/twoSum.cpp
#include <iostream>
#include <bits/stdc++.h>
#include <vector>
#include <unordered_map>

using namespace std;

void twoSum(vector<int> &nums, int target){
    unordered_map<char,int> counts;
    for(int i=0;i<nums.size();i++){
        counts[nums[i]]++;
    }
    for(int i=0;i<nums.size();i++){
        counts[nums[i]]--;
        int anVal = target - nums[i];
        if(counts[anVal] > 0){
            for(int j=i;j<nums.size();j++){
                if(anVal == nums[j]){
                    cout << i << " " << j << endl;
                }
            }
        }
    }
};
int main(int argc, char **argv){
    vector<int> nums = {2,7,11,15};
    int target = 9;
    twoSum(nums,target);
    return 0;
}